compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/test/conv_to_im2col.mlir

```diff
@@ -195,3 +195,53 @@ util.func public @conv_2d_nhwc_fhwc(%arg0: tensor<1x16x16x4xf32>, %arg1: tensor<
 // CHECK-SAME:   iterator_types = ["parallel", "parallel", "parallel", "parallel", "reduction"]
 // CHECK-SAME:   ins(%[[IM2COL]], %[[COLLAPSED]] : tensor<1x14x14x36xf32>, tensor<16x36xf32>)
 // CHECK:      util.return %[[MATMUL]] : tensor<1x14x14x16xf32>
+
```

Comment:
Would you plan to create a test to also cover conv3d cases or would you plan to do that in a later PR.

BTW, I think we may need a standalone test just for conv2d but with mis-ordered conv2d case.

Reply:
I was thinking about it and decided no to not add too many tests, since the implementation is already ND and the logic is not doing anything different based on the number of dims, What do you think? I did test it locally though. The test I am adding covers the transpostions that can happen on the filter and permutation maps.

Reply:
I adapted an existing generic conv test to have transpose in the map.

Reply:
Thanks!